src/pages-and-resources/teams/TeamSetEditor.jsx

```diff
@@ -0,0 +1,174 @@
+import FormikErrorFeedback from '../../generic/FormikErrorFeedback';
+import messages from './messages';
+
+// Maps a team type to its corresponding intl message
+const TeamTypeNameMessage = {
+  [TeamSetTypes.OPEN]: {
+    label: messages.teamSetTypeOpen,
+    description: messages.teamSetTypeOpenDescription,
+  },
+  [TeamSetTypes.PUBLIC_MANAGED]: {
+    label: messages.teamSetTypePublicManaged,
+    description: messages.teamSetTypePublicManagedDescription,
+  },
+  [TeamSetTypes.PRIVATE_MANAGED]: {
+    label: messages.teamSetTypePrivateManaged,
+    description: messages.teamSetTypePrivateManagedDescription,
+  },
+};
+
+function TeamSetEditor({
+  intl, teamSet, onDelete, onChange, onBlur, fieldNameCommonBase, errors,
+}) {
+  const [isDeleting, setDeleting] = useState(false);
+  const [isOpen, setOpen] = useState((teamSet.id === null) || true);
+  const initiateDeletion = () => setDeleting(true);
+  const cancelDeletion = () => setDeleting(false);
+
+  const handleToggle = (open) => setOpen(Boolean(errors.name || errors.maxTeamSize || errors.description) || open);
+
+  const formGroupClasses = 'mb-4';
+
+  return (
+    <TransitionReplace>
+      {isDeleting
+        ? (
+          <div className="d-flex flex-column card rounded mb-3 px-3 py-2 p-4" key="isDeleting">
+            <h3>{intl.formatMessage(messages.teamSetDeleteHeading)}</h3>
+            <p>{intl.formatMessage(messages.teamSetDeleteBody)}</p>
+            <div className="d-flex flex-row justify-content-end">
+              <Button variant="muted" size="sm" onClick={cancelDeletion}>
+                {intl.formatMessage(messages.cancel)}
+              </Button>
+              <Button variant="outline-danger" size="sm" onClick={() => onDelete(teamSet)}>
+                {intl.formatMessage(messages.delete)}
+              </Button>
+            </div>
+          </div>
+        )
+        : (
+          <CollapsableEditor
+            // If there is no id, this is a new team, so automatically open it
+            key="isConfiguring"
+            open={isOpen}
+            onToggle={handleToggle}
+            onDelete={initiateDeletion}
+            deleteAlt={intl.formatMessage(messages.deleteAlt)}
+            expandAlt={intl.formatMessage(messages.expandAlt)}
+            collapseAlt={intl.formatMessage(messages.collapseAlt)}
+            title={
+              isOpen
+                ? (
+                  <div className="d-flex flex-column flex-shrink-1 h4 p-0 m-0">
+                    {intl.formatMessage(messages.configureTeamSet)}
+                  </div>
+                ) : (
+                  <div className="d-flex flex-column small flex-shrink-1">
+                    <span>{intl.formatMessage(TeamTypeNameMessage[teamSet.type].label)}</span>
+                    <span className="h4 text-truncate">{teamSet.name || '<new>'}</span>
+                    <span className="text-truncate">{teamSet.description || '<new>'}</span>
```

Comment:
I think in case of new now we don't need this check as it's not collapsible so we can remove`|| '<new>'`